haas/cli.py

```diff
@@ -441,68 +628,169 @@ def port_detach_nic(switch, port):
 @cmd
 def list_free_nodes():
     """List all free nodes"""
-    url = object_url('free_nodes')
-    do_get(url)
+
+    try:
+        ret = h.list_free_nodes()
+        print "SUCCESS: Total free nodes available with HaaS: %s " % len(ret())
+        for i in ret():
+            print " %s" % i
+
+    except DuplicateName as e:
+        print e.message
+    except IncompleteDependency as e:
+        print e.message
+    except UnknownError as e:
+        print e.message
 
 @cmd
 def list_project_nodes(project):
     """List all nodes attached to a <project>"""
-    url = object_url('project', project, 'nodes')
-    do_get(url)
+
+    try:
+        ret = h.list_project_nodes(project)
+        print "SUCCESS: Total nodes allocated to Project %s : %s " % (project,len(ret()))
+        for i in ret():
+            print " %s" % i
+    except DuplicateName as e:
+        print e.message
+    except IncompleteDependency as e:
+        print e.message
+    except UnknownError as e:
+        print e.message
 
 @cmd
 def list_project_networks(project):
     """List all networks attached to a <project>"""
-    url = object_url('project', project, 'networks')
-    do_get(url)
+
+    try:
+        ret = h.list_project_networks(project)
+        print "SUCCESS: Total networks with Project %s: %s " % (project,len(ret()))
+        for i in ret():
```

Comment:
I think you mean `ret` not `ret()`.

Also - for these types of functions, I'd be fine with dumping the json. What do others think?